src/middleware.js

```diff
@@ -65,6 +66,7 @@ function handlePromise(dispatch, getState, action) {
     });
     handleEventHook(meta, 'onFailure', error, getState);
     handleEventHook(meta, 'onFinish', false, getState);
+    return { success: false, payload: error };
```

Comment:
After playing around with this locally, I wonder if `return Promise.reject(error)` would be better here? That way you could keep they `then/catch` semantics. (Or does that suggest that `dispatch` failed, rather than the promise?)